Api/src/main/java/playlist/server/ranking/controller/RankingController.java

```diff
@@ -0,0 +1,69 @@
+package playlist.server.ranking.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import playlist.server.mainpagerankingInfo.vo.MainPageRankingInfoVo;
+import playlist.server.ranking.service.RankingService;
+
+@RestController
+@RequestMapping("/ranking")
+@RequiredArgsConstructor
+@Tag(name = "1. [랭킹]")
+public class RankingController {
+
+    private final RankingService rankingService;
+
+    @Operation(summary = "랭킹을 조회합니다.")
+    @GetMapping("/daily")
+    public ResponseEntity<MainPageRankingInfoVo> getDailyRanking(
+            @RequestParam(name = "type", defaultValue = "like") String type
+    ) {
+        try {
+            MainPageRankingInfoVo rankingInfoVo = rankingService.getRanking("daily", type, "boardId");
+            if (rankingInfoVo == null) {
+                return ResponseEntity.notFound().build();
+            }
+            return ResponseEntity.ok(rankingInfoVo);
+        } catch (Exception e) {
+            return ResponseEntity.badRequest().build();
+        }
+    }
+
+    @Operation(summary = "주간 랭킹을 조회합니다.")
+    @GetMapping("/weekly")
+    public ResponseEntity<MainPageRankingInfoVo> getWeeklyRanking(
+            @RequestParam(name = "type", defaultValue = "like") String type
+    ) {
+        try {
+            MainPageRankingInfoVo rankingInfoVo = rankingService.getRanking("weekly", type, "boardId");
+            if (rankingInfoVo == null) {
+                return ResponseEntity.notFound().build();
+            }
+            return ResponseEntity.ok(rankingInfoVo);
+        } catch (Exception e) {
+            return ResponseEntity.badRequest().build();
+        }
```

Comment:
다른메소드 동일하지만 service레벨에서 Exception을 내주고 있는것 같아서 badRequest Return의 경우에는 발생시키는 이유를 모르겠네요.

코드를 이런식으로 만들어 볼 수 있지않을까요?
```suggestion
MainPageRankingInfoVo rankingInfoVo = rankingService.getRanking(RankingInfo.WEEKLY, type, "boardId");
if (rankingInfoVo == null) {
    return ResponseEntity.notFound().build();
}
return ResponseEntity.ok(rankingInfoVo);
```

Reply:
두번째 인자값인 type의 경우에도 enum으로 정의가 가능할 것 같아요.
세번째 인자값인 boardId는 차라리 어딘가에 `public static String`으로 변수선언한 이후에 해당 값을 모두 공유해서 사용하면
추후 변경이 발생시 편하게 공유할 수 있지 않을까요?

Reply:
boardId 이거는 못했습니다..두번째 인자값인 type의 경우에도 enum으로 정의가 가능할 것 같아요. 이건 했습니다!